app/components/select/item.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react"
+import { ActivityIndicator, View } from "react-native"
+
+import { ListItem, ListItemProps, makeStyles, useTheme } from "@rneui/themed"
+import Icon from "react-native-vector-icons/Ionicons"
+
+const useStyles = makeStyles(() => ({
+  iconContainer: { width: 18 },
+}))
+
+export type ItemProps = {
+  children: React.ReactNode
+  value: string
+  selected: boolean
+  onChange: (optionKey: string) => void | Promise<void>
+  loading: boolean
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>
+} & ListItemProps
+
+export const Item: React.FC<ItemProps> = ({
+  children,
+  value,
+  selected,
+  onChange,
+  loading,
+  setLoading,
+  ...props
+}) => {
+  const {
+    theme: { colors },
+  } = useTheme()
+  const styles = useStyles()
+
+  const [showActivityIndicator, setShowActivityIndicator] = React.useState(false)
+
+  const onPress = () => {
+    if (selected || loading) return
+
+    const isPromise = onChange(value)
+    if (isPromise) {
+      isPromise
+        .then(() => {
```

Comment:
we don't use `then` in our codebase. 
you can use async/await instead. it will make the code more readable (may not be too obvious when there is one level of nesting, but it's become very apparent where there is several level of nesting.)